flake.nix

```diff
@@ -27,51 +27,34 @@
           overlays = [
             haskellNix.overlay
             (final: prev: {
-              hapistrano-ghc8107 = final.haskell-nix.cabalProject' {
+              hapistrano-ghc966 = final.haskell-nix.cabalProject' {
                 src = final.haskell-nix.haskellLib.cleanGit {
                   name = "hapistrano";
                   src = ./.;
                 };
-                compiler-nix-name = "ghc8107";
-              };
-              hapistrano-ghc902 = final.haskell-nix.cabalProject' {
-                src = final.haskell-nix.haskellLib.cleanGit {
-                  name = "hapistrano";
-                  src = ./.;
+                # This is used by `nix develop .` to open a shell for use with
+                # `cabal`, `hlint` and `haskell-language-server`
+                shell.tools = {
+                  cabal = {};
+                  hlint = {};
+                  haskell-language-server = {};
                 };
-                compiler-nix-name = "ghc902";
+                compiler-nix-name = "ghc966";
```

Comment:
@DavidMazarro If we plan to use only one version of GHC to compile the project moving forward, I believe we could refactor this further:

```nix
{
  ...
  outputs = { self, nixpkgs, flake-utils, haskellNix }:
    flake-utils.lib.eachSystem [ ... ] (system:
    let
      overlays = [ haskellNix.overlay
        (final: prev: {
          # This overlay adds our project to pkgs
          hapistrano =
            final.haskell-nix.project' {
              src = ./.;
              compiler-nix-name = "ghc966";
              # This is used by `nix develop .` to open a shell for use with
              # `cabal`, `hlint` and `haskell-language-server`
              shell.tools = {
                cabal = {};
                hlint = {};
                haskell-language-server = {};
              };
              # Non-Haskell shell tools go here
              shell.buildInputs = with pkgs; [
                nixpkgs-fmt
              ];
              # This adds `js-unknown-ghcjs-cabal` to the shell.
              # shell.crossPlatforms = p: [p.ghcjs];
            };
        })
      ];
      pkgs = import nixpkgs { inherit system overlays; inherit (haskellNix) config; };
      flake = pkgs.hapistrano.flake {
        # This adds support for `nix build .#js-unknown-ghcjs:hello:exe:hello`
        # crossPlatforms = p: [p.ghcjs];
      };
    in flake // rec {
      # Built by `nix build .`
      packages= {
        default = flake.packages."hapistrano:exe:hap";
        test = flake.packages."hapistrano:test:test".overrideAttrs(...);
      };
      apps = {
        type = "app";
        program = "${packages.test}/bin/test";
      }
    });
}
```

Source: https://input-output-hk.github.io/haskell.nix/tutorials/getting-started-flakes.html#scaffolding

Reply:
Good point. Yeah, I was following that same documentation when I added the shell tools 😄